### ISSUE TITLE
negative predictions ?

### ISSUE THREAD

**Issue by Vicktoria13** [zexianghui/DenseFlow_localization#2]
Hello,

i tried to launch the test code on vigor dataset, and the line :
flow_predictions, flow_conf, mask, pre_u, pre_v, pre_theta = net(sat, grd, meter_per_pixel, \
                                                        torch.zeros_like(gt_shift_u,device=gt_shift_u.device), \
                                                        torch.zeros_like(gt_shift_v,device=gt_shift_v.device), \
                                                        torch.zeros_like(gt_heading,device=gt_heading.device), mode='train', \
                                                        file_name=0)

is giving negative pre_u, pre_v, pre_theta. Do they represent a shift ? How exactly do you deduce and reproduce the plot with the discrete position of the camera into the satellite image ?

thanks !

**Comment by XianghuiZe** [comment]
Hi! Thanks for your interest in our work.

We set pre_u, pre_v, and pre_theta to zero matrices because we don’t want the model to have access to ground-truth information during testing. In this context, torch.zeros_like(gt_shift_u, device=gt_shift_u.device) doesn’t carry any practical meaning—it’s just a placeholder :)

Our pipeline projects the ground-view image into BEV, then performs point matching between the BEV image and the satellite map, and finally estimates rotation and translation using a least-squares solution. For more details, please refer to our paper: https://proceedings.neurips.cc/paper_files/paper/2023/file/df5f94d6ac6e13d830d70536cde9f0d2-Paper-Conference.pdf
